Emscripten-build-instructions.md

```diff
@@ -0,0 +1,167 @@
+llvm build. This can be done by executing the following
+
+```bash
+export LLVM_BUILD_DIR=$PWD
+```
+
+We can move onto building the wasm  
+version of CppInterOp. To do this execute the following  
+
+```bash
+cd ../../CppInterOp/
+mkdir build
+cd ./build/
+emcmake cmake -DCMAKE_BUILD_TYPE=Release    \
+                -DUSE_CLING=OFF                             \
+                -DUSE_REPL=ON                               \
+                -DLLVM_DIR=$LLVM_BUILD_DIR/lib/cmake/llvm      \
+                -DLLD_DIR=$LLVM_BUILD_DIR     \
+                -DClang_DIR=$LLVM_BUILD_DIR/lib/cmake/clang     \
+                -DBUILD_SHARED_LIBS=ON                      \
+                -DCMAKE_FIND_ROOT_PATH_MODE_PACKAGE=ON            \
+                -DCMAKE_INSTALL_PREFIX=$PREFIX         \
+                ../
+emmake make -j $(nproc --all)
+```bash
+
+Once this finishes building we need to take note of where we built our  
+llvm build. This can be done by executing the following
+
+```bash
+export CPPINTEROP_BUILD_DIR=$PWD
+```
+
+## Xeus-cpp-lite Wasm Build Instructions
+
+A project which makes use of the wasm build of CppInterOp is xeus-cpp.  
+xeus-cpp is a C++ Jupyter kernel. To build xeus-cpp we first need to  
+install several other dependencies. To create an Conda environment  
+with this dependencies installed execute the following from the  
+CppInterOp build folder (these instructions will assume you have  
+already installed micromamba)  
+
+```bash
+cd ..
+micromamba create -f environment-wasm.yml --platform=emscripten-wasm32
+micromamba activate CppInterOp-wasm
+export PREFIX=$MAMBA_ROOT_PREFIX/envs/CppInterOp-wasm
+```
+Now we can build an wasm build of xeus-cpp by executing the following  
+
+```bash
+cd ..
+git clone --depth=1 https://github.com/compiler-research/xeus-cpp.git
+cd ./xeus-cpp
+mkdir build
+cd build
+export CMAKE_PREFIX_PATH=$PREFIX
+export CMAKE_SYSTEM_PREFIX_PATH=$PREFIX 
+emcmake cmake \
+          -DCMAKE_BUILD_TYPE=Release                                     \
+          -DCMAKE_PREFIX_PATH=$PREFIX                                    \
+          -DCMAKE_INSTALL_PREFIX=$PREFIX                                 \
+          -DXEUS_CPP_EMSCRIPTEN_WASM_BUILD=ON                            \
+          -DCMAKE_FIND_ROOT_PATH_MODE_PACKAGE=ON                         \
+          -DCppInterOp_DIR="$CPPINTEROP_BUILD_DIR/lib/cmake/CppInterOp"  \
+          ..
+ emmake make -j $(nproc --all) install
+```
+
+To build Jupyter Lite with this kernel without creating a website you can  
+execute the following
+
+```bash
+cd ../..
+micromamba create -n xeus-lite-host jupyterlite-core
+micromamba activate xeus-lite-host
+python -m pip install jupyterlite-xeus
+jupyter lite build --XeusAddon.prefix=$PREFIX
+```
+
+Once the Jupyter Lite site has built you can test the website locally  
+by executing
+
+```bash
+jupyter lite serve --XeusAddon.prefix=$PREFIX
+```
```

Comment:
I don't think there is any need to paste the xeus-cpp build again. Just provide a link from xeus-cpp's master for the same. This would ensure we have one place to go to when interested and the changes also need to be done at one place only.

Reply:
We should just mention if cppinterop is built from source then we need to provide this config 
`-DCppInterOp_DIR="$CPPINTEROP_BUILD_DIR/lib/cmake/CppInterOp"  \`

Reply:
I plan to keep these instructions here rather than point to xeus-cpp instructions. This is because the instructions are slightly different if testing xeus-cpp against a CppInterOp built from source.